vllm/attention/layer.py

```diff
@@ -649,8 +646,10 @@ def unified_attention(
 
     forward_context: ForwardContext = get_forward_context()
     attn_metadata = forward_context.attn_metadata
-    if isinstance(attn_metadata, dict):
+    if isinstance(attn_metadata, dict) and layer_name in attn_metadata:
         attn_metadata = attn_metadata[layer_name]
+    else:
+        attn_metadata = None
```

Comment:
I think that there are times where there's only a single value of attention metadata so it's not actually per-layer. In those cases we still don't want it to be None, so I think we can simplify like this:
```suggestion
    if isinstance(attn_metadata, dict):
        attn_metadata = attn_metadata.get(layer_name)
```

Reply:
I think that's handled by the get method which will return none for those layers